fineract-core/src/main/java/org/apache/fineract/commands/service/CommandSourceService.java

```diff
@@ -64,8 +66,16 @@ public CommandSource saveInitialSameTransaction(CommandWrapper wrapper, JsonComm
 
     @NotNull
     private CommandSource saveInitial(CommandWrapper wrapper, JsonCommand jsonCommand, AppUser maker, String idempotencyKey) {
-        CommandSource initialCommandSource = getInitialCommandSource(wrapper, jsonCommand, maker, idempotencyKey);
-        return commandSourceRepository.saveAndFlush(initialCommandSource);
+        try {
+            CommandSource initialCommandSource = getInitialCommandSource(wrapper, jsonCommand, maker, idempotencyKey);
+            return commandSourceRepository.saveAndFlush(initialCommandSource);
+        } catch (JpaSystemException jse) {
+            final String message = (jse.getRootCause() != null) ? jse.getRootCause().getMessage() : null;
+            if (message != null && message.contains("UNIQUE_PORTFOLIO_COMMAND_SOURCE")) {
```

Comment:
I reckon we need to slightly modify to work with MariaDB and Postgres as well:
`message.toLowerCase().contains("unique_portfolio_command_source")`

Reply:
Done!